### ISSUE TITLE
TinyGPEmulator returns NaN predictions on banana_2d under seed 0 (default jitter)

### ISSUE THREAD

**Issue by arob5** [arob5/sabi#52]
## Summary

`TinyGPEmulator` with default `jitter=1e-3` produces a NaN-laden Cholesky factor (and therefore NaN predictive means everywhere) on `banana_2d` under PRNG seed 0 with `n_initial=8, n_rounds=8` and `ExpectedImprovement` acquisition. Switching to `jitter=1e-1` fixes it; other seeds (1–7) work with the default.

This was uncovered while authoring the Getting Started notebook (PR #47 / #50). The notebook works around it by switching to seed 3, but the underlying numerical instability is a real issue worth tracking.

## Reproduction

```python
import jax
import jax.numpy as jnp
import numpy as np
from probpipe import mean
from sabi.acquisitions.ei import ExpectedImprovement
from sabi.algorithms import Algorithm, run
from sabi.emulators import TinyGPEmulator
from sabi.metrics.mmd import MMD
from sabi.problems.benchmarks import banana_2d

jax.config.update("jax_enable_x64", True)

algorithm = Algorithm(
    emulator_factory=lambda: TinyGPEmulator(input_shape=(2,)),  # default jitter
    acquisition=ExpectedImprovement(),
    n_initial=8, n_rounds=8,
    metrics=(MMD(n_estimate_samples=512, n_reference_samples=512),),
)
result = run(banana_2d(), algorithm, jax.random.key(0))

# Predictive mean at training points is NaN:
m = np.asarray(mean(result.emulator(result.X)))
assert np.isfinite(m).sum() == 0  # all NaN
```

## What I observed

- `cache.alpha` (the precomputed `K^{-1} y` vector): 13/15 entries finite, 2 NaN.
- `cache.L_sigma` (Cholesky of `K + (noise+jitter) I`): 222/225 entries finite, 3 NaN.
- `Xs_train` and `Ys_train` are fully finite.
- Min pairwise distance between the 15 training points is 0.55 in original units (no near-duplicates).
- Y range is `[-361, -1.8]` — large dynamic range, but `ZScoreScaler` standardizes Y before fit, so the GP works on order-1 values.

The NaN is therefore being introduced during Cholesky factorization, not during data prep.

## Sweep across seeds

| seed | predictive finite (3 grid pts) | final MMD² |
|-----:|-------------------------------:|-----------:|
| 0    | 0/3 ❌                          | 0.4159     |
| 1    | 3/3 ✅                          | 0.9069     |
| 2    | 3/3 ✅                          | 0.4885     |
| 3    | 3/3 ✅                          | 0.4078     |
| 4    | 3/3 ✅                          | 0.7679     |
| 5    | 3/3 ✅                          | 0.4799     |
| 6    | 3/3 ✅                          | 0.4740     |
| 7    | 3/3 ✅                          | 0.5666     |

## Sweep across jitter (seed = 0)

| jitter | predictive finite |
|-------:|------------------:|
| 1e-3 (default) | 0/3 ❌ |
| 1e-2 | 0/3 ❌ |
| 1e-1 | 3/3 ✅ |

## Likely cause

`tinygp` Cholesky on the standardized inputs produces a matrix that is poorly conditioned for this specific seed-0 design. The default `jitter=1e-3` (added to the diagonal as `noise + jitter` per `_TinyGPCache.build`) isn't enough to recover positive-definiteness in this case, and the resulting NaN propagates through `alpha = solve(L, Ys)` and downstream into `predict_latent`.

## Suggested directions

- **Bump the default `jitter`** to something like `1e-2` (which still failed for seed 0 above, so possibly higher) or even `1e-1`. Trade-off: too much jitter biases the predictive mean toward the prior near training points; this needs a sanity check on accuracy.
- **Adaptive jitter** — retry Cholesky with progressively larger jitter until it succeeds. tinygp doesn't ship this out of the box, but it's a few lines in `_TinyGPCache.build`.
- **Check for NaN at fit time and raise** a clear exception ("Cholesky produced NaN; consider increasing jitter via `TinyGPEmulator(..., jitter=1e-2)`"). At minimum, fail loudly instead of silently returning NaN-laden predictions that propagate to estimates and metrics.
- **Investigate the seed-0 design** specifically — is the initial 8-point `LatinHypercube` (or whatever `Algorithm` uses by default) producing a configuration that's pathologically aligned with the Matérn 5/2 kernel? Understanding *why* seed 0 fails when 1–7 succeed might point at a structural fix.

## Reference

Discovered in PR #50 (fix for the blank surrogate panel in Getting Started). That PR works around the issue by switching the notebook seed; this issue tracks the underlying fix.
